src/RequestParser.cpp

```diff
@@ -0,0 +1,724 @@
+	if (contentLength != length) {
+		request.errorCode = StatusBadRequest;
+		throw std::runtime_error(ERR_CONTENT_LENGTH);
+	}
+}
+
+/**
+ * @brief Checks the validity of an HTTP header field name.
+ *
+ * This function verifies if the provided HTTP header field name is valid
+ * according to HTTP specifications.
+ * It ensures that the header name does not end with whitespace and that all
+ * characters in the header name are valid according to `isValidHeaderFieldNameChar`.
+ *
+ * @param headerName The HTTP header field name to be validated.
+ * @param request The HTTP request object to be filled.
+ * @throws std::runtime_error If the header name contains invalid characters or
+ *         ends with whitespace, an error is thrown and the error code is set to StatusBadRequest.
+ */
+void RequestParser::checkHeaderName(const std::string& headerName, HTTPRequest& request)
+{
+	if (isspace(headerName[headerName.size() - 1]) != 0) {
+		request.errorCode = StatusBadRequest;
+		throw std::runtime_error(ERR_HEADER_COLON_WHITESPACE);
+	}
+	for (size_t i = 0; i < headerName.size(); i++) {
+		if (!isValidHeaderFieldNameChar(headerName[i])) {
+			request.errorCode = StatusBadRequest;
+			throw std::runtime_error(ERR_HEADER_NAME_INVALID_CHAR);
+		}
+	}
+}
+
+/**
+ * @brief Checks the validity of the Content-Length header field.
+ *
+ * This function verifies the validity of the Content-Length header field in an HTTP request.
+ * It ensures that:
+ * - There is only one Content-Length header or multiple with the same value.
+ * - The value of Content-Length is a valid numeric string.
+ * If the header is invalid or there are multiple Content-Length headers with conflicting values,
+ * an appropriate error code is set, and a `std::runtime_error` is thrown.
+ *
+ * @param headerName The name of the HTTP header field.
+ * @param headerValue The value associated with the HTTP header field.
+ * @param request The HTTP request object to be filled.
+ * @throws std::runtime_error If there are multiple conflicting Content-Length
+ *         headers or if the Content-Length value is invalid, an error is thrown
+ *         and the error code is set to StatusBadRequest.
+ */
+void RequestParser::checkContentLength(const std::string& headerName, std::string& headerValue, HTTPRequest& request)
+{
+	if (headerName == "Content-Length") {
+		if (headerValue.empty()) {
+			request.errorCode = StatusBadRequest;
+			throw std::runtime_error(ERR_INVALID_CONTENT_LENGTH);
+		}
+		if (request.headers.find("Content-Length") != request.headers.end()
+			&& request.headers["Content-Length"] != headerValue) {
+			request.errorCode = StatusBadRequest;
+			throw std::runtime_error(ERR_MULTIPLE_CONTENT_LENGTH_VALUES);
+		}
+
+		std::vector<std::string> strValues = webutils::split(headerValue, ", ");
+		std::vector<long> numValues;
+		for (size_t i = 0; i < strValues.size(); i++) {
+			char* endptr = NULL;
+			long contentLength = strtol(strValues[i].c_str(), &endptr, decimalBase);
```

Comment:
Could use std namespace version: std::strtol
long contentLength can be made const

Reply:
Changed.